### PR TITLE
Enhance overview components and add agent creation metadata

### PR DESCRIPTION
**Pull request by rasika2012**
## Purpose
image
image

Issue: https://github.com/wso2/agent-manager/issues/1352

This pull request adds support for tracking and exposing the creator of an agent ("createdBy") in the Agent Manager Service API. The change is best-effort: it attempts to resolve the user who created the agent from audit information, and exposes this in API responses when available. The implementation includes model changes, OpenAPI spec updates, service logic, and comprehensive tests.

### API and Model Changes

* Added a new `createdBy` field to the `AgentResponse` model and OpenAPI spec, representing the user who created the agent, including both user ID and (if available) display name/username. [[1]](diffhunk://#diff-a032eb87e30830102f49b75ccb3f928590c6b37cf9af746044321fa65c5be43eR11450-R11451) [[2]](diffhunk://#diff-a032eb87e30830102f49b75ccb3f928590c6b37cf9af746044321fa65c5be43eR11461-R11480) [[3]](diffhunk://#diff-976596bf5839f836dfb156feff762f84104f238a3c8e8ef8728a9825df0189e8R42-R56) [[4]](diffhunk://#diff-7b8840bdfc7d87c73ea431ddcbced02be541d0b93c318eb2db702b4ac037d00bR40) [[5]](diffhunk://#diff-7b8840bdfc7d87c73ea431ddcbced02be541d0b93c318eb2db702b4ac037d00bR452-R483) [[6]](diffhunk://#diff-7b8840bdfc7d87c73ea431ddcbced02be541d0b93c318eb2db702b4ac037d00bR520-R522) [[7]](diffhunk://#diff-f73f95785ac3d0a0372f664b74bfb0b1fd9174652d8de2716bcdbc3ce6815765R1-R153)

### Service Implementation

* Implemented logic in `agentManagerService` to best-effort resolve and populate the `createdBy` field using the Thunder identity store, handling cases where the information is missing or the user has been deleted.
* Updated service dependencies and constructor to wire in the new identity client. [[1]](diffhunk://#diff-103dc40ba45c9fc03cdb8f0c32376d76421b325930fe8e929b4f368e1d5cafdbR89) [[2]](diffhunk://#diff-103dc40ba45c9fc03cdb8f0c32376d76421b325930fe8e929b4f368e1d5cafdbR108) [[3]](diffhunk://#diff-103dc40ba45c9fc03cdb8f0c32376d76421b325930fe8e929b4f368e1d5cafdbR123)

### Testing

* Added comprehensive unit tests for the new `populateCreatedBy` logic, covering all edge cases (no provisioning, no identity client, errors, missing data, deleted users, etc.).
* Extended test stubs to support the new identity view method. [[1]](diffhunk://#diff-5e514cf8eb6e806658563e239b5234953b0758aa5ab4252c0d81f294dc038656R51) [[2]](diffhunk://#diff-5e514cf8eb6e806658563e239b5234953b0758aa5ab4252c0d81f294dc038656R74-R77)

### Utilities

* Added a utility function to map internal agent creator info to the API wire type for responses.

### Codebase Maintenance

* Added a `moq` code generation directive for mocking the `IdentityClient` interface, improving testability.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

 

## Summary by CodeRabbit

* **New Features**
  * Agent details now show creator IDs and resolved display names when available.
  * Added clickable overview cards for identities, configurations, capabilities, and related resources.
  * Added reusable page headers, metadata displays, avatars, and overview cards.

* **Improvements**
  * Refreshed agent overview layout, labels, metrics, evaluations, and observability messaging.
  * Added clearer creation and update metadata across provider and proxy pages.
  * Improved navigation, copyable Agent IDs, deployment links, and status indicators.
  * Reduced unnecessary notifications for background data requests.
  * Renamed navigation and identifier labels for clarity.

### CODE REVIEW

File: agent-manager-service/services/agent_manager.go [hunk 1]
```diff
@@ -1037,10 +1040,61 @@ func (s *agentManagerService) GetAgent(ctx context.Context, ouID string, project
...
 
+// populateCreatedBy best-effort resolves and attaches who created this agent.
+// The only signal available is the audit-only requester id captured once at
+// agent-creation time on every environment's AgentID binding (see
+// models.AgentThunderClient.RequestedBy) — not a purpose-built "creator"
+// column, since the agent record itself lives in OpenChoreo, not AMS's DB.
+// Never fails GetAgent: a missing provisioning implementation, a lookup
+// error, an empty requestedBy (pre-dates this being tracked), or a
+// since-deleted user all just leave CreatedBy unset (or id-only).
+func (s *agentManagerService) populateCreatedBy(ctx context.Context, ouID, projectName, agentName string, agent *models.AgentResponse) {
```
Comment (jhivandb) [inline agent-manager-service/services/agent_manager.go:1050]:
```suggestion
// populateCreatedBy best-effort resolves and attaches who created this agent.
func (s *agentManagerService) populateCreatedBy(ctx context.Context, ouID, projectName, agentName string, agent *models.AgentResponse) {
```
Shall we shrink this

**Review by jhivandb — APPROVED** [review]
